server.go

```diff
@@ -2104,10 +2104,22 @@ func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User, ch
 		var i int
 		var stmt influxql.Statement
 		for i, stmt = range q.Statements {
-			// Set default database and policy on the statement.
-			if err := s.NormalizeStatement(stmt, database); err != nil {
-				results <- &Result{Err: err}
-				break
+			// If a default database wasn't passed in by the caller,
+			// try to get it from the statement.
+			defaultDB := database
+			if defaultDB == "" {
+				if s, ok := stmt.(influxql.HasDefaultDatabase); ok {
+					defaultDB = s.DefaultDatabase()
```

Comment:
Not sure I understand but I'll take a shot.  Currently, only `CreateContinuousQueryStatement` implements this method and it returns the `Database` member.  The parser will error if the database is not specified.  If another statement were to implement it later (which will probably happen) and `Database` were an optional field or the default database for that statement was a logical value based on conditions, it could return an empty string, indicating that the default database wasn't set.